nilearn/decomposition/single_pca.py

```diff
@@ -0,0 +1,320 @@
+        See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
+        List of subject data
+
+    mask_img: Niimg-like object
+        See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
+        Mask to apply on the data
+
+    parameters: dictionary
+        Dictionary of parameters passed to `filter_and_mask`. Please see the
+        documentation of the `NiftiMasker` for more informations.
+
+    confounds: CSV file path or 2D matrix
+        This parameter is passed to signal.clean. Please see the
+        corresponding documentation for details.
+
+    n_components: integer, optional
+        Number of components to be extracted by the PCA
+
+    memory_level: integer, optional
+        Integer indicating the level of memorization. The higher, the more
+        function calls are cached.
+
+    memory: joblib.Memory
+        Used to cache the function calls.
+
+    verbose: integer, optional
+        Indicate the level of verbosity (0 means no messages).
+
+    copy: boolean, optional
+        Whether or not data should be copied
+
+    random_state: int or RandomState
+        Pseudo number generator state used for random sampling.
+
+    random_state:
+    """
+
+    data, affine = cache(
+        filter_and_mask, memory,
+        func_memory_level=2, memory_level=memory_level,
+        ignore=['verbose', 'memory', 'memory_level', 'copy'])(
+            imgs, mask_img, parameters,
+            memory_level=memory_level,
+            memory=memory,
+            verbose=verbose,
+            confounds=confounds,
+            copy=copy)
+    if n_components <= data.shape[0] // 4:
+        U, S, _ = cache(randomized_svd, memory, memory_level=memory_level,
+                        func_memory_level=2)(
+            data.T, n_components, random_state=random_state)
+    else:
+        U, S, _ = cache(linalg.svd, memory, memory_level=memory_level,
+                        func_memory_level=2)(
+            data.T, full_matrices=False)
+    U = U.T[:n_components].copy()
+    S = S[:n_components]
+    return U, S
+
+
+class SinglePCA(BaseEstimator, CacheMixin):
+    """Perform Single Subject Principal Component Analysis. Perform a PCA on each subject and stack the results
+
+    Parameters
+    ----------
+    n_components: int
+        Number of components to extract
+
+    smoothing_fwhm: float, optional
+        If smoothing_fwhm is not None, it gives the size in millimeters of the
+        spatial smoothing to apply to the signal.
+
+    mask: Niimg-like object, instance of NiftiMasker or MultiNiftiMasker, optional
+        Mask to be used on data. If an instance of masker is passed,
+        then its mask will be used. If no mask is given,
+        it will be computed automatically by a MultiNiftiMasker with default
+        parameters.
+
+    do_cca: boolean, optional
+        Indicate if a Canonical Correlation Analysis must be run after the
+        PCA.
```

Comment:
In the context of this object, no CCA is run. It's more that the subject-level data is whitened, for instance as a preparation step to do a CCA.